src/libraries/Common/src/Interop/Browser/Interop.Runtime.cs

```diff
@@ -0,0 +1,625 @@
+        [MethodImplAttribute(MethodImplOptions.InternalCall)]
+        internal static extern object New(string className, object[] _params, out int exceptional_result);
+        [MethodImplAttribute(MethodImplOptions.InternalCall)]
+        internal static extern object TypedArrayToArray(int js_obj_handle, out int exceptional_result);
+        [MethodImplAttribute(MethodImplOptions.InternalCall)]
+        internal static extern object TypedArrayCopyTo(int js_obj_handle, int array_ptr, int begin, int end, int bytes_per_element, out int exceptional_result);
+        [MethodImplAttribute(MethodImplOptions.InternalCall)]
+        internal static extern object TypedArrayFrom(int array_ptr, int begin, int end, int bytes_per_element, int type, out int exceptional_result);
+        [MethodImplAttribute(MethodImplOptions.InternalCall)]
+        internal static extern object TypedArrayCopyFrom(int js_obj_handle, int array_ptr, int begin, int end, int bytes_per_element, out int exceptional_result);
+
+        // / <summary>
+        // / Execute the provided string in the JavaScript context
+        // / </summary>
+        // / <returns>The js.</returns>
+        // / <param name="str">String.</param>
+        public static string InvokeJS(string str)
+        {
+            var res = InvokeJS(str, out int exception);
+            if (exception != 0)
+                throw new JSException(res);
+            return res;
+        }
+
+        public static Interop.JavaScript.Function? CompileFunction(string snippet)
+        {
+            var res = CompileFunction(snippet, out int exception);
+            if (exception != 0)
+                throw new JSException((string)res);
+            return res as Interop.JavaScript.Function;
+        }
+
+        private static Dictionary<int, JSObject?> bound_objects = new Dictionary<int, JSObject?>();
+        private static Dictionary<object, JSObject?> raw_to_js = new Dictionary<object, JSObject?>();
+
+        static Runtime()
+        { }
+
+        public static int New<T>(params object[] _params)
+        {
+            var res = New(typeof(T).Name, _params, out int exception);
+            if (exception != 0)
+                throw new JSException((string)res);
+            return (int)res;
+        }
+
+        public static int New(string hostClassName, params object[] _params)
+        {
+            var res = New(hostClassName, _params, out int exception);
+            if (exception != 0)
+                throw new JSException((string)res);
+            return (int)res;
+        }
+
+        public static JSObject? NewJSObject(JSObject? js_func_ptr = null, params object[] _params)
+        {
+            var res = NewObjectJS(js_func_ptr?.JSHandle ?? 0, _params, out int exception);
+            if (exception != 0)
+                throw new JSException((string)res);
+            return res as JSObject;
+        }
+
+        private static int BindJSObject(int js_id, Type mappedType)
```

Comment:
Not sure these methods should be private

Reply:
Most of these are called from the javascript side and can really be anything.  Had to add an accessor to pass build so kept it private.  Will change them to internal.

Reply:
Addressed in recent commit